camera_calibration/nodes/cameracalibrator.py

```diff
@@ -113,17 +122,24 @@ def main():
 
     options, args = parser.parse_args()
 
-    if len(options.size) != len(options.square):
+    if (len(options.size) != len(options.square)) or (options.pattern == "charuco" and (
```

Comment:
Please break this logic out into a new function `or self.CharucoValid()` or something so that its not so convoluted

Reply:
:+1: